torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1083,12 +1098,19 @@ c10::intrusive_ptr<ProcessGroup::Work> ProcessGroupNCCL::collective(
   if (sequenceNum_) {
     sequenceNum_->increment();
   }
+
+  // Inputs must either all be on different devices,
+  // or all be on different devices.
```

Comment:
I don't quite get this comment, it says inputs must all be on different devices or all be on different devies?

Reply:
lol oops, that should definitely say "all be on different devices, or all be on the same device"